crates/js-sys/src/lib.rs

```diff
@@ -463,6 +503,152 @@ extern "C" {
+        #[wasm_bindgen(static_method_of = Atomics, catch)]
+        pub fn load(typed_array: &JsValue, index: u32) -> Result<f64, JsValue>;
+
+        /// The static `Atomics.notify()` method notifies up some agents that
+        /// are sleeping in the wait queue.
+        /// Note: This operation works with a shared `Int32Array` only.
+        ///
+        /// [MDN documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Atomics/notify)
+        #[wasm_bindgen(static_method_of = Atomics, catch)]
+        pub fn notify(typed_array: &JsValue, index: u32, count: u32) -> Result<u32, JsValue>;
+
+        /// The static `Atomics.or()` method computes a bitwise OR with a given value
+        /// at a given position in the array, and returns the old value at that position.
+        /// This atomic operation guarantees that no other write happens
+        /// until the modified value is written back.
+        ///
+        /// [MDN documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Atomics/or)
+        #[wasm_bindgen(static_method_of = Atomics, catch)]
+        pub fn or(typed_array: &JsValue, index: u32, value: f64) -> Result<f64, JsValue>;
+
+        /// The static `Atomics.store()` method stores a given value at the given
+        /// position in the array and returns that value.
+        ///
+        /// [MDN documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Atomics/store)
+        #[wasm_bindgen(static_method_of = Atomics, catch)]
+        pub fn store(typed_array: &JsValue, index: u32, value: f64) -> Result<f64, JsValue>;
+
+        /// The static `Atomics.sub()` method substracts a given value at a
+        /// given position in the array and returns the old value at that position.
+        /// This atomic operation guarantees that no other write happens
+        /// until the modified value is written back.
+        ///
+        /// [MDN documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Atomics/sub)
+        #[wasm_bindgen(static_method_of = Atomics, catch)]
+        pub fn sub(typed_array: &JsValue, index: u32, value: f64) -> Result<f64, JsValue>;
+
+        /// The static `Atomics.wait()` method verifies that a given
+        /// position in an `Int32Array` still contains a given value
+        /// and if so sleeps, awaiting a wakeup or a timeout.
+        /// It returns a string which is either "ok", "not-equal", or "timed-out".
+        /// Note: This operation only works with a shared `Int32Array`
+        /// and may not be allowed on the main thread.
+        ///
+        /// [MDN documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Atomics/wait)
+        #[wasm_bindgen(static_method_of = Atomics, catch)]
+        pub fn wait(typed_array: &JsValue, index: u32, value: i32) -> Result<JsValue, JsValue>;
```

Comment:
@alexcrichton it works only with `Int32Array`. Is it fine to keep `value` type as `i32`?

Reply:
Yeah if this is spec'd as only supporting a `Int32Array` I think it's fine to use an `i32` here. Should the `Ok` value of the `Result` returned also be `i32`?

Reply:
It returns "ok" | "not-equal" | "timed-out". Should it be `String` or `JsString` instead?

Reply:
Well then, that's news to me! In that case yeah let's change this to return `JsString`